### PR TITLE
applications: nrf_desktop/machine_learning: Align CONFIG_RESET_ON_FATAL_ERROR in child image configs

### PR DESCRIPTION
**Pull request by mkapala-nordic**
Jira: NCSDK-22426

### CODE REVIEW

**Review by kapi-no — COMMENTED** [review]
Please add changelog entries

File: applications/nrf_desktop/configuration/nrf52833dk_nrf52833/child_image/mcuboot/prj.conf [hunk 1]
```diff
@@ -64,3 +64,6 @@ CONFIG_BOOT_BANNER=n
 
 # Use minimal C library instead of the Picolib
 CONFIG_MINIMAL_LIBC=y
+
+# Improve debugging experience by disabling reset on fatal error
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/configuration/nrf52833dk_nrf52833/child_image/mcuboot/prj.conf:68]: btw. I would also consider aligning this for B0 bootloader and HCI rpmsg configurations (either under a separate tickets or we can expand scope and story points of this ticket)

**Review by MarekPieta — CHANGES_REQUESTED** [review]

**Comment by mkapala-nordic** [comment]
Only rebased

File: applications/nrf_desktop/configuration/nrf52833dk_nrf52833/child_image/mcuboot/prj.conf (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkapala-nordic) [reply applications/nrf_desktop/configuration/nrf52833dk_nrf52833/child_image/mcuboot/prj.conf:68]: Applied, please verify if i did not miss any conf file as there are a few of them :D

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -201,7 +201,10 @@ nRF5340 Audio
 nRF Machine Learning (Edge Impulse)
 -----------------------------------
 
-|no_changes_yet_note|
+* Updated the MCUboot and ``hci_rpmsg`` child image debug configurations to disable the :kconfig:option:`CONFIG_RESET_ON_FATAL_ERROR` Kconfig option.
```
Comment (MarekPieta) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:204]: `MCUboot and HCI RMPsg child images` -> to improve consistency
Also in other places

File: applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/child_image/mcuboot/prj_release.conf [hunk 3]
```diff
@@ -63,3 +63,6 @@ CONFIG_BOOT_SERIAL_IMG_GRP_HASH=n
 
 # Use minimal C library instead of the Picolib
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/child_image/mcuboot/prj_release.conf:64]: typo in commit message (both commits):
`explicitely` -> `explicitly`

**Review by MarekPieta — APPROVED** [review]

**Review by zycz — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -220,6 +223,10 @@ nRF Desktop
     The ``CONFIG_USB_HID_PROTOCOL_CODE`` Kconfig option is deprecated and a dedicated API needs to be used instead.
   * The :ref:`nrf_desktop_watchdog` to use ``watchdog0`` DTS alias instead of ``wdt`` DTS node label.
     Using the alias makes the configuration of the module more flexible.
+  * The MCUboot, B0 and HCI RPMsg child images debug configurations to disable the :kconfig:option:`CONFIG_RESET_ON_FATAL_ERROR` Kconfig option.
+    Disabling this Kconfig option improves the debugging experience.
+  * The MCUboot, B0 and HCI RPMsg child images release configurations to explicitly enable the :kconfig:option:`CONFIG_RESET_ON_FATAL_ERROR` Kconfig option.
+    Enabling this Kconfig option improves the reliability of the firmware.
```
Comment (annwoj) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:229]:
```suggestion
  * The MCUboot, B0, and HCI RPMsg child images debug configurations to disable the :kconfig:option:`CONFIG_RESET_ON_FATAL_ERROR` Kconfig option.
    Disabling this Kconfig option improves the debugging experience.
  * The MCUboot, B0, and HCI RPMsg child images release configurations to explicitly enable the :kconfig:option:`CONFIG_RESET_ON_FATAL_ERROR` Kconfig option.
    Enabling this Kconfig option improves the reliability of the firmware.
```

**Comment by mkapala-nordic** [comment]
Applied @annwoj suggestions

**Comment by mkapala-nordic** [comment]
Rebased

**Review by annwoj — APPROVED** [review]

**Comment by mkapala-nordic** [comment]
Rebased

**Pull request merged**